apps/rush-lib/src/data/RushConfiguration.ts

```diff
@@ -145,16 +145,19 @@ export default class RushConfiguration {
    * Loads the configuration data from an Rush.json configuration file and returns
    * an RushConfiguration object.
    */
-  public static loadFromConfigurationFile(rushJsonFilename: string): RushConfiguration {
+  public static loadFromConfigurationFile(
+    rushJsonFilename: string,
+    validateVersion: boolean = true
```

Comment:
>validateVersion: boolean = true [](start = 4, length = 31)

This parameter is nonsensical:

1. When invoked from the command line, the version of rush-lib always matches the version of the config file (for now)

2. But some tool is using rush-lib to read rush.json, there is no such expectation.

The config file format does not change every time we publish rush-lib, so rush-lib needs to be compatible with a range of versions, always, regardless of who called this API.

Reply:
Maybe the criteria should be:

rush.json cannot be newer than rush-lib itself

rush.json cannot be older than some "oldest supported version" that should generally be pretty ancient

---
In reply to: [147529670](https://github.com/Microsoft/web-build-tools/pull/417#discussion_r147529670) [](ancestors = 147529670)